src/chunked_scatter/safe_scatter.py

```diff
@@ -0,0 +1,244 @@
+    """
+    Scatter the regions into chunks. All chunks will be of size
+    min_scatter_size, except (possibly) the last region.
+    The last region will be >= min_scatter_size < min_scatter_size*2
+    """
+    # Make sure we don't get a floating minimum scatter size
+    min_scatter_size = int(min_scatter_size)
+
+    if min_scatter_size < 1:
+        raise RuntimeError("min_scatter_size must be a positive integer")
+
+    for region in regions:
+        # How much of the region is still remaining
+        remaining = len(region)
+
+        # If the current region is so small we cannot get 2 min_scatter_size
+        # from it, just yield the entire region and continue
+        if remaining < 2*min_scatter_size:
+            yield region
+            continue
+
+        # We keep looping as long as there are at least 2 min_scatter_size left
+        # of the region
+        while remaining >= 2*min_scatter_size:
+            contig, start, end = region
+            new_region = BedRegion(contig, start, start+min_scatter_size)
+            yield new_region
+            region = BedRegion(contig, start+min_scatter_size, end)
+            remaining = len(region)
+        yield BedRegion(contig, start+min_scatter_size, end)
+
+
+def safe_scatter(regions: List[BedRegion],
+                 scatter_count: int,
+                 min_scatter_size: int = 10000,
+                 mix: bool = False,
+                 ) -> Generator[List[BedRegion], None, None]:
+    """
+    Scatter the regions equally over the specified scatter_count.
+
+    The terminology here is that we will generate scatter_count "bins", and we
+    will place each of the scattered regions (len >= min_scatter_size) in one
+    of the bins.
+
+    This is done in a manner that guarantees the following:
+      - No regions will be split smaller than min_scatter_size. There can be
+        regions smaller than min_scatter_size, but only when the original
+        the original region was smaller than min_scatter_size.
+      - All scatters will be within min_scatter_size of each other.
+
+    Any overlapping bed records are merged together.
+    :param regions: The regions over which to scatter.
+    :param scatter_count: The number of bins to create.
+    :param min_scatter_size: The minimum size of a scattered region.
+    allowed to be split across multiple lists.
+    :return: Yields lists of BedRegions which can be converted into bed files.
+    """
+    # What is the target size for the bins?
+    target_bin_size = determine_bin_size(regions, scatter_count)
+
+    if target_bin_size < min_scatter_size:
+        msg = (f"--min-scatter-size is not compatible with the provided "
+               f"regions and number of bins ({min_scatter_size} > "
+               f"{target_bin_size})")
+        raise RuntimeError(msg)
+
+    # Mix small and regular regions
+    if mix:
+        regions = mix_small_regions(regions, target_bin_size)
```

Comment:
According to codecov there is no test for mix==True. That should be added.

Reply:
Added tests with mix=true